### PR TITLE
ICU-22723 CLDR 46 alpha2 to main

### PR DESCRIPTION
**Pull request by DraganBesevic**
##### Checklist

- [ ] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22723
- [ ] Required: The PR title must be prefixed with a JIRA Issue number. 
- [ ] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [ ] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [ ] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

Alpha 2 integration for CLDR 46 that changes since alpha 0 (note that we skipped alpha 1).

This change includes updates for Likely Subtag algorithm and data, which caused some known issues

https://unicode-org.atlassian.net/browse/CLDR-17908
https://unicode-org.atlassian.net/browse/ICU-22869

Unit test are updated where CLDR data changed, as in number formatting for en-ZA and plural rules for scn.

ALLOW_MANY_COMMITS=true

### CODE REVIEW

**Review by srl295 — APPROVED** [review]

File: icu4c/source/test/cintltst/cloctst.c [hunk 1]
```diff
@@ -6202,6 +6202,11 @@ static void TestLikelySubtags(void)
         const char* const minimal = full_data[i][0];
         const char* const maximal = full_data[i][1];
 
+		if (uprv_strcmp(minimal, "und_Hant_CN") == 0 &&
```
Comment (markusicu) [inline icu4c/source/test/cintltst/cloctst.c:6205]: Please go through the source code changes and replace tabs with spaces.

**Review by markusicu — CHANGES_REQUESTED** [review]
mostly rslgtm

I see the collation change for ₨ shining through :-)

File: icu4c/source/test/intltest/caltest.cpp [hunk 2]
```diff
@@ -5634,8 +5634,10 @@ void CalendarTest::TestFirstDayOfWeek() {
     // ssh => ssh_Arab_AE => Saturday
     verifyFirstDayOfWeek("ssh", UCAL_SATURDAY);
     // wbl_Arab => wbl_Arab_AF => Saturday
-    verifyFirstDayOfWeek("wbl-Arab", UCAL_SATURDAY);
-
+	if (!logKnownIssue("CLDR-17907", "wbl-Arab returns wrong first day of the week, probably caused by wrong Likely Subtag algo")) {
+       verifyFirstDayOfWeek("wbl-Arab", UCAL_SATURDAY);
+    }
```
Comment (markusicu) [inline icu4c/source/test/intltest/caltest.cpp:5639]: I agree with [Mark's ticket comment](https://unicode-org.atlassian.net/browse/ICU-22869?focusedCommentId=176427). Please remove this test case altogether.

File: icu4c/source/test/cintltst/cloctst.c (hunk 1 above)
Comment (DraganBesevic) [reply icu4c/source/test/cintltst/cloctst.c]: I think I caught them all. Sorry!

File: icu4c/source/test/intltest/caltest.cpp (hunk 2 above)
Comment (DraganBesevic) [reply icu4c/source/test/intltest/caltest.cpp]: Gone

**Review by markusicu — APPROVED** [review]
thanks!

**Pull request merged**